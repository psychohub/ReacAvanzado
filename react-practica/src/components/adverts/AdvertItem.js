import React from 'react';
import { Link } from 'react-router-dom';
import {
  MDBCard,
  MDBCardBody,
  MDBCardTitle,
  MDBCardText,
  MDBCardImage,
  MDBCol,
} from 'mdb-react-ui-kit';

const AdvertItem = ({ advert }) => {
  return (
    <MDBCol md="4">
      <MDBCard>
        <MDBCardBody>
          <MDBCardTitle>
            <Link to={`/adverts/${advert.id}`}>{advert.name}</Link>
          </MDBCardTitle>
          {advert.photo && (
            <MDBCardImage
              src={advert.photo}
              alt={advert.name}
              style={{
                maxWidth: '100%',
                height: '200px',
                objectFit: 'cover',
                marginBottom: '1rem',
              }}
            />
          )}
          <MDBCardText>
            <span>Precio: {advert.price}</span>
            <br />
            <span>Tipo: {advert.sale ? 'Venta' : 'Compra'}</span>
            <br />
            <span>Tags: {advert.tags.join(', ')}</span>
          </MDBCardText>
        </MDBCardBody>
      </MDBCard>
    </MDBCol>
  );
};

export default AdvertItem;
