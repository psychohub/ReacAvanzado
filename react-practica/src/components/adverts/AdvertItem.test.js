import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AdvertItem from './AdvertItem';

jest.mock('react-router-dom', () => {
  const originalModule = jest.requireActual('react-router-dom');
  return {
    ...originalModule,
    Link: ({ children }) => children,
  };
});

describe('AdvertItem', () => {
  it('DeberÃ­a renderizarse correctamente', () => {
    const advert = {
      id: 1,
      name: 'Advert 1',
      price: 100,
      sale: true,
      tags: ['tag1', 'tag2'],
    };

    const { container } = render(
      <MemoryRouter>
        <AdvertItem advert={advert} />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
