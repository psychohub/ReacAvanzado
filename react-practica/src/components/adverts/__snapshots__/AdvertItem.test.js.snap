// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvertItem Deber√≠a renderizarse correctamente 1`] = `
<div>
  <div
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Advert 1
        </h5>
        <p
          class="card-text"
        >
          <span>
            Precio: 
            100
          </span>
          <br />
          <span>
            Tipo: 
            Venta
          </span>
          <br />
          <span>
            Tags: 
            tag1, tag2
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
